src/App.tsx

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGetAll = () => {
+    getAll()
+      .then(setGoods)
+      .catch(error => setErrorMessage(error.message));
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGet5First = () => {
+    get5First()
+      .then(setGoods)
+      .catch(error => setErrorMessage(error.message));
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const handleGetRedGoods = () => {
+    getRedGoods()
+      .then(setGoods)
+      .catch(error => setErrorMessage(error.message));
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const visibleGoods = useMemo(() => {
+    return goods;
+  }, [goods]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={handleGetAll}
+      >
+        Load all goods
+      </button>
+      {errorMessage && <p style={{ color: 'red' }}>{errorMessage}</p>}
```

Comment:
but if i will have an error - I will see 3 errors
![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/91411692/fb6563f6-c3e3-4830-bb0d-38f309e0e7fe)